src/controllers/presentation.ts

```diff
@@ -0,0 +1,247 @@
+						return false;
+					}
+				}
+				return true;
+			})
+			.withMessage('An invalid JWT string'),
+		check('holderDid').optional().isString().withMessage('Invalid holder DID'),
+		check('verifierDid').optional().isString().withMessage('Invalid verifier DID'),
+		check('policies').optional().isObject().withMessage('Verification policies should be an object'),
+	];
+
+	public static presentationVerifyValidator = [
+		check('presentation')
+			.exists()
+			.withMessage('W3c verifiable presentation was not provided')
+			.custom((value) => {
+				if (typeof value === 'string' || typeof value === 'object') {
+					return true;
+				}
+				return false;
+			})
+			.withMessage('Entry must be a JWT or a presentation body with JWT proof')
+			.custom((value) => {
+				if (typeof value === 'string') {
+					try {
+						jwtDecode(value);
+					} catch (e) {
+						return false;
+					}
+				}
+				return true;
+			})
+			.withMessage('An invalid JWT string'),
+		check('verifierDid').optional().isString().withMessage('Invalid verifier DID'),
+		check('policies').optional().isObject().withMessage('Verification policies should be an object'),
+		check('makeFeePayment').optional().isBoolean().withMessage('makeFeePayment: should be a boolean').bail(),
+		query('verifyStatus').optional().isBoolean().withMessage('verifyStatus should be a boolean value'),
+	];
+
+	/**
+	 * @openapi
+	 *
+	 * /presentation/create:
+	 *   post:
+	 *     tags: [ Presentation ]
+	 *     summary: "!!! WARN. Such endpoint is made mostly for testing purposes and it is not supposed to be used in production !!! Create a Verifiable Presentation from credential(s)."
+	 *     description: "This endpoint creates a Verifiable Presentation from credential(s). As input, it can take the credential(s) as a string or the entire credential(s) itself. \n !!! WARN. Such endpoint is made only for testing purposes !!!"
+	 *     requestBody:
+	 *       content:
+	 *         application/x-www-form-urlencoded:
+	 *           schema:
+	 *             $ref: '#/components/schemas/PresentationCreateRequest'
+	 *         application/json:
+	 *           schema:
+	 *             $ref: '#/components/schemas/PresentationCreateRequest'
+	 *     responses:
+	 *       200:
+	 *         description: The request was successful.
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               $ref: '#/components/schemas/PresentationCreateResult'
+	 *       400:
+	 *         $ref: '#/components/schemas/InvalidRequest'
+	 *       401:
+	 *         $ref: '#/components/schemas/UnauthorizedError'
+	 *       500:
+	 *         $ref: '#/components/schemas/InternalError'
+	 */
+
+	public async createPresentation(request: Request, response: Response) {
+		const result = validationResult(request);
+		if (!result.isEmpty()) {
+			return response.status(StatusCodes.BAD_REQUEST).json({ error: result.array()[0].msg });
+		}
+
+		const { credential, holderDid, verifierDid } = request.body;
```

Comment:
we should allow multiple credentials here

Reply:
Agreed